From a5446a7aac318f977c18b55a9353bcd08a390b72 Mon Sep 17 00:00:00 2001
From: Aleksander Morgado <aleksander@aleksander.es>
Date: Mon, 6 Jun 2016 20:42:20 +0200
Subject: [PATCH 09/19] libqmi-glib,mbim: don't create mbim device multiple
 times

---
 src/libqmi-glib/qmi-device.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/libqmi-glib/qmi-device.c b/src/libqmi-glib/qmi-device.c
index 775fe1d..0eccb1b 100644
--- a/src/libqmi-glib/qmi-device.c
+++ b/src/libqmi-glib/qmi-device.c
@@ -1939,6 +1939,15 @@ create_mbim_device (DeviceOpenContext *ctx)
 {
     GFile *file;
 
+    if (ctx->self->priv->mbimdev) {
+        g_simple_async_result_set_error (ctx->result,
+                                         QMI_CORE_ERROR,
+                                         QMI_CORE_ERROR_WRONG_STATE,
+                                         "Already open");
+        device_open_context_complete_and_free (ctx);
+        return;
+    }
+
     g_debug ("[%s] creating MBIM device...", ctx->self->priv->path_display);
     file = g_file_new_for_path (ctx->self->priv->path);
     mbim_device_new (file,
-- 
2.11.0

